kernel/param.h

```diff
@@ -11,3 +11,4 @@
 #define NBUF         (MAXOPBLOCKS*3)  // size of disk block cache
 #define FSSIZE       2000  // size of file system in blocks
 #define MAXPATH      128   // maximum file path name
+#define NPROC_KSTACK  80   // number of kstack for procs to allocate on boot
```

Comment:
I've been testing out some numbers and on 80 alloctest passes but it failes on 100. So naturally we want to make this number as big as possible and since I knew 80 was something of a maximum value (I haven't tested it this much I don't know if alloctest passes on 90 for example) I just used it.

Reply:
Maybe fail allocproc when there are all 80 procs are runnings? What stack will 81th proc use?

Reply:
A, it will use newly allocated.

Reply:
It seems for me that this number can be even 0 as we have kstack allocation on demand

Reply:
Sure and it was so but I actually claim that using this constant is better. Basicly if we allocate on demand and not reusing allocated memory - it is memory leak. I don't know how to free the memory allocated with kvmmap function so I used preallocated buffer to minimize this memory leak and it actually shows (if you run usertests at the end it will fail due to som memory leaks but you can the number of free pages left - for this implementation it is around 31000 out of 32000). I will definetly research kvmmap internal structure more today as a part of lab3 journey